app/Models/Plan/PlanDay.php

```diff
@@ -89,6 +89,20 @@ public function verifyDayCompleted()
         ];
     }
 
+    /**
+     * Validate if the playlist attached to the current day has filesets attached.
+     *
+     * @return bool
+     */
+    public function hasContentAvailable(Playlist $playlist_to_eval = null) : bool
+    {
+        if (!is_null($playlist_to_eval) && $this['playlist_id'] === $playlist_to_eval->id) {
+            return isset($playlist_to_eval->items) ? sizeof($playlist_to_eval->items) > 0 : false;
+        }
+
+        return PlaylistItems::where('playlist_id', $this['playlist_id'])->count() > 0;
```

Comment:
this is going to calculate the count (expensive), then only consider if it is zero or greater than zero. It would be faster if was changed to be like the other optimized queries (using Exists, which returns true earlier)

Reply:
It has been updated to use the EXISTS operator otherwise COUNT as you can see in the next line:

https://github.com/faithcomesbyhearing/dbp/pull/684/files#r776118010